From 1c95210a7135eb707073431d34f1e9739233dbc9 Mon Sep 17 00:00:00 2001
From: Ronny Lorenz <ronny@tbi.univie.ac.at>
Date: Sun, 20 Aug 2017 00:46:21 +0200
Subject: [PATCH 20/33] tegra_soctherm: make middle operand explicit

---
 drivers/thermal/tegra_soctherm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/thermal/tegra_soctherm.c b/drivers/thermal/tegra_soctherm.c
index c449b43ccdc5..3ff5dbeff958 100644
--- a/drivers/thermal/tegra_soctherm.c
+++ b/drivers/thermal/tegra_soctherm.c
@@ -1406,7 +1406,7 @@ static int soctherm_init_platform_data(struct platform_device *pdev)
 		therm = &plat_data->therm[tsensor2therm_map[i]];
 		s = &plat_data->sensor_data[i];
 		s->sensor_enable = s->zone_enable;
-		s->sensor_enable = s->sensor_enable ?: therm->zone_enable;
+		s->sensor_enable = s->sensor_enable ? s->sensor_enable : therm->zone_enable;
 	}
 
 	/* Pdiv */
-- 
2.22.0

